src/chart/generateCategoricalChart.tsx

```diff
@@ -1103,6 +1104,69 @@ export const generateCategoricalChart = ({
         mouseHandlerCallback: this.triggeredAfterMouseMove,
         layout: this.props.layout,
       });
+      this.displayDefaultTooltip();
+    }
+
+    displayDefaultTooltip() {
+      const { defaultIndex } = this.props;
+      if (
+        typeof this.props.defaultIndex !== 'number' ||
```

Comment:
any particular reason for the number check? TypeScript should take care of things

Reply:
To protect against runtime errors from consumers who are using .jsx, not .tsx. I can imagine someone passing a `null` or something in the prop, and I don't want this method to choke and kill the whole chart.

Reply:
JS users still see types and typehints, imo if they specify something that should be a number as a string that is on them for the most part. But all good to leave the checks